### PR TITLE
compiler: arcmwdt: Remove unsupported compile flags (-fno-pic and -fno-pie)

### PR DESCRIPTION
**Pull request by kokas-a**
Clang version used by ARCMWDT does not support -fno-pic and -fno-pie flags.
Flags were added into arcmwdt branch by
'commit 8259931fce1b ("xcc-clang: Do not used unavailable options")'. Initially they were set in common CMakeLists.txt via zephyr_cc_option() function, which filtered them out, and they did not used in builing process.

Signed-off-by: Nikolay Agishev <agishev@synopsys.com>

### CODE REVIEW

File: cmake/compiler/arcmwdt/compiler_flags.cmake [hunk 1]
```diff
@@ -175,11 +175,8 @@ set_property(TARGET compiler-cpp PROPERTY required "-Hcplus" "-Hoff=Stackcheck_a
 # Compiler flag for turning off thread-safe initialization of local statics
 set_property(TARGET compiler-cpp PROPERTY no_threadsafe_statics "-fno-threadsafe-statics")
 
-#Compiler flags for disabling position independent code / executable
-set_compiler_property(PROPERTY no_position_independent
-                      -fno-pic
-                      -fno-pie
-)
+# Clang version used by ARCMWDT does not support -fno-pic and -fno-pie
```
Comment (evgeniy-paltsev) [inline cmake/compiler/arcmwdt/compiler_flags.cmake:178]: Please add comment that PIE is disabled by default for ARCMWDT.

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

File: cmake/compiler/arcmwdt/compiler_flags.cmake [hunk 2]
```diff
@@ -175,11 +175,9 @@ set_property(TARGET compiler-cpp PROPERTY required "-Hcplus" "-Hoff=Stackcheck_a
 # Compiler flag for turning off thread-safe initialization of local statics
 set_property(TARGET compiler-cpp PROPERTY no_threadsafe_statics "-fno-threadsafe-statics")
 
-#Compiler flags for disabling position independent code / executable
-set_compiler_property(PROPERTY no_position_independent
-                      -fno-pic
-                      -fno-pie
-)
+# Clang version used by ARCMWDT does not support -fno-pic and -fno-pie flags
+# PIE is disabled by default for ARCMWDT
```
Comment (evgeniy-paltsev) [inline cmake/compiler/arcmwdt/compiler_flags.cmake:179]: I'm not sure it's about clang version. I'd rephrased it to something like that:

```
# ARC MWDT does not support -fno-pic and -fno-pie flags, but it has PIE is disabled by default - so we don't need any extra flags here.
```

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

**Review by evgeniy-paltsev — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
